### PR TITLE
Add explicit skip

### PR DESCRIPTION
**Pull request by HaoK**

### CODE REVIEW

**Comment by HaoK** [comment]
cc @Tratcher

File: src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs [hunk 1]
```diff
@@ -336,7 +336,7 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
                 }
                 else if (messageReceivedContext.Skipped)
                 {
-                    return AuthenticateResult.Success(ticket: null);
+                    return AuthenticateResult.Skipped();
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs]: For this to work you need to update https://github.com/aspnet/Security/blob/dev/src/Microsoft.AspNet.Authentication/RemoteAuthenticationHandler.cs#L25 as well, correct?

File: src/Microsoft.AspNet.Authentication/AuthenticateResult.cs [hunk 2]
```diff
@@ -54,5 +54,9 @@ public static AuthenticateResult Failed(string errorMessage)
             return new AuthenticateResult() { Error = new Exception(errorMessage) };
         }
 
+        public static AuthenticateResult Skipped()
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Authentication/AuthenticateResult.cs]: Add a WasSkipped property that checks Ticket == null && Error == null?

File: src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs (hunk 1 above)
Comment (HaoK) [inline src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs]: Why?  What does it mean to skip and produce no ticket?  That's still an error wrt to the base... We have nothing to return

Comment (HaoK) [inline src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs]: I still don't understand what it means for everything to skip and noone produces a ticket...that feels like an error to me :)

**Comment by PinpointTownes** [comment]
Could we also get a `Failed` overload taking no parameter? It would be used to indicate that authentication failed while bypassing the automatic logging (to log the error separately, for instance).

**Comment by Tratcher** [comment]
Replaced

**Pull request closed without merging**